api/http/handler/endpoints/endpoint_create.go

```diff
@@ -66,6 +67,11 @@ func (payload *endpointCreatePayload) Validate(r *http.Request) error {
 	}
 	payload.EndpointCreationType = endpointCreationEnum(endpointCreationType)
 
+	payload.ContainerEngine, err = request.RetrieveMultiPartFormValue(r, "ContainerEngine", true)
+	if err != nil || (payload.ContainerEngine != portainer.ContainerEngineDocker && payload.ContainerEngine != portainer.ContainerEnginePodman && payload.ContainerEngine != "") {
+		return errors.New("invalid container engine value. Value must be one of: 'docker' or 'podman'")
+	}
+
```

Comment:
I'd introduce default value handling here just to make sure we don't get into weird situations later when handling other values. This also maps the payload with the behavior of the handler.

```go
if payload.ContainerEngine == "" {
  payload.ContainerEngine = portainer.ContainerEngineDocker
}
```

Reply:
updated